clients/fides-js/src/lib/initialize.ts

```diff
@@ -342,32 +351,81 @@ export const initialize = async ({
+      /**
+       * Now that we've determined the effective PrivacyExperience, update it
+       * with some additional client-side state so that it is initialized with
+       * the user's current consent preferences, etc. and ready to display!
+       */
+      const updatedExperience = updateExperience({
         cookie,
         experience: effectiveExperience,
         debug: options.debug,
         isExperienceClientSideFetched: fetchedClientSideExperience,
       });
-      debugLog(options.debug, "Updated experience", updatedExperience);
+      debugLog(
+        options.debug,
+        "Updated experience from saved preferences",
+        updatedExperience
+      );
       Object.assign(effectiveExperience, updatedExperience);
+
+      /**
+       * Finally, update the "cookie" state to track the user's *current*
+       * consent preferences as determined by the updatedExperience above. This
+       * "cookie" state is then published to external listeners via the
+       * Fides.consent object and Fides events like FidesInitialized below, so
+       * we rely on keeping it up to date!
+       *
+       * DEFER (PROD-1780): This is quite *literally* duplicate state, and means
+       * we end up with three potential sources of consent preferences:
+       * 1. "savedConsent": user's *saved* consent from the fides_consent cookie
+       * 2. "experience.privacy_notices[].current_preference": user's current/unsaved preferences shown in the UI
+       * 3. "cookie": another version of user's current/unsaved preferences
+       *
+       * Simplifying this state management helps us simplify FidesJS, but it's
+       * tricky to do without accidentally breaking something, so be careful!
+       */
+      const updatedCookie = updateCookieFromExperience({
+        cookie,
+        experience: effectiveExperience,
+      });
+      debugLog(
+        options.debug,
+        "Updated current cookie state from experience",
+        updatedCookie
+      );
+      Object.assign(cookie, updatedCookie);
+
       if (shouldInitOverlay) {
         await initOverlay({
           experience: effectiveExperience,
           fidesRegionString: fidesRegionString as string,
           cookie,
+          savedConsent,
           options,
           renderOverlay,
         }).catch(() => {});
       }
+
+      /**
+       * Last up: apply GPC to the current preferences automatically. This will
+       * set any applicable notices to "opt-out" unless the user has previously
+       * saved consent, etc.
+       *
+       * NOTE: Do *not* await the results of this function, even though it's
+       * async and returns a Promise! Instead, let the GPC update run
+       * asynchronously but continue our initialization. If GPC applies, this
+       * will kick off an update to the user's consent preferences which will
+       * also call the Fides API, but we want to finish initialization
+       * immediately while those API updates happen in parallel.
```

Comment:
Nota bene!

This is another piece of complexity; I was going to change this to `await automaticallyApplyGPCPreferences(...)` but don't want to introduce a new `await` here since it'd change the order of initialization...! Woof.